File: Countries/js/6.js
// There is at least one country name that contains another country name. 
// Find all of these cases.

let main = () => {

    const fs = require("fs");
    let buffer_arry = fs.readFileSync("Countries/countries.txt", "utf-8");
    let in_file = buffer_arry.split("\n");
    let country_list = [];
    
    // Function creates a list of all countries with whitespace removed
    let create_list = () => {

        for(let i = 0; i < in_file.length; i++) {
            country_list.unshift(in_file[i].toLowerCase());
        };  
        return;
    };

    // Function compares country names
    let compare_words = () => {

        for(let i = 0; i < in_file.length; i++) {
            let word = in_file[i].toLowerCase();
            for(let j = 0; j < country_list.length; j++) {

                   if(word != country_list[j] && 
                    word.includes(country_list[j]) == true) {
                       console.log(country_list[j], "-IN-", word);
                   };  
            };
        };

    };

    create_list();
    compare_words();

};

main();

File: Countries/python/6.py
# There is at least one country name that contains another country name. Find all of these cases.
# Create two list to compare against each other. Lower and strip whitespace from words.

def main():

    in_file = open("../countries.txt", "r")
    word_list = []
    for words in in_file:
        word_list.insert(0, words.lower().strip())

    for words in in_file:
        for i in range(len(word_list)):
            words = words.lower().strip()
            if(words != word_list[i]):
                if(words in word_list[i]):
                    print(words,"-IN-", word_list[i])

if(__name__ == "__main__"):
    main()
        